src/compressed_tensors/transforms/random_hadamard.py

```diff
@@ -86,13 +86,11 @@ def __init__(
             permutation_name=permutation_name,
         )
 
-        if not self.matrix_registry.contains(size):
-            self.matrix_registry.set_matrix(self.size, self.transform)
-
-    def fetch(self):
+    def fetch(self, dtype, device):
         transform = self.matrix_registry.get_matrix(self.size)
         if transform is None:
-            transform = random_hadamard_matrix(size=self.size)
+            transform = random_hadamard_matrix(size=self.size).to(dtype).to(device)
+            self.matrix_registry.set_matrix(self.size, transform)
```

Comment:
It might be nice to abstract this one layer down, so the transform just calls `random_hadamard_matrix` or `deterministic_hadamard_matrix` with `use_cache=True`, and the function automatically checks the registry/cache and returns the same value if a matrix of the same size and arguments has already been generated